### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by jacob-berger**
Still need to add tests for UserController and GroupController

### CODE REVIEW

**Comment by kenny2892** [comment]
I will review this.

File: SecretSanta/test/SecretSanta.Api.Tests/GiftControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        [TestMethod]
+        public async Task GetItemWithId_Success()
+        {
+            var service = new Mock<IGiftService>();
+            service.Setup(service => service.FetchByIdAsync(12)).Returns(Task.FromResult(SampleData.CreateN64Gift()));
+            var giftController = new GiftController(service.Object);
+
+            ActionResult<Gift> result = await giftController.Get(12);
+
+            Assert.IsInstanceOfType(result.Result, typeof(OkObjectResult));
```
Comment (kenny2892) [inline SecretSanta/test/SecretSanta.Api.Tests/GiftControllerTests.cs]: You should also check the returned gift to make sure it is the correct gift.

File: SecretSanta/test/SecretSanta.Api.Tests/GiftControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        public async Task Put_Success()
+        {
+            var service = new Mock<IGiftService>();
+            Gift gift = SampleData.CreateN64Gift();
+            service.Setup(service => service.UpdateAsync(12, gift)).Returns(Task.FromResult<Gift?>(gift));
+            var giftController = new GiftController(service.Object);
+
+            ActionResult<Gift> result = await giftController.Put(12, gift);
+
+            Assert.IsNotNull(result);
```
Comment (kenny2892) [inline SecretSanta/test/SecretSanta.Api.Tests/GiftControllerTests.cs]: You should also check that the result is an OkObjectResult and that the result had the updated gift.

File: SecretSanta/test/SecretSanta.Api.Tests/GiftControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        }
+
+        [TestMethod]
+        public async Task Delete_Success()
+        {
+            var service = new Mock<IGiftService>();
+            var gift = SampleData.CreateN64Gift();
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Api.Tests/GiftControllerTests.cs]: Local variable 'gift' should be used

File: SecretSanta/Directory.Build.props [hunk 4]
```diff
@@ -2,6 +2,7 @@
   <PropertyGroup>
     <TargetFramework>netcoreapp3.1</TargetFramework>
     <Nullable>enable</Nullable>
+    <IncludeOpenAPIAnalyzers>true</IncludeOpenAPIAnalyzers>
```
Comment (Keboo) [inline SecretSanta/Directory.Build.props]: This certainly woks, but outside of web projects it does not add any value

File: SecretSanta/src/SecretSanta.Api/Controllers/GiftController.cs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                return NotFound();
+            }
+        }
+
+        [HttpDelete("{id}")]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<IActionResult> Delete(int id)
+        {
+            if (await GiftService.DeleteAsync(id) == true)
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Api/Controllers/GiftController.cs]:
```suggestion
            if (await GiftService.DeleteAsync(id))
```

**Pull request closed without merging**